core/src/main/scala/kafka/metrics/KafkaMetricsReporter.scala

```diff
@@ -50,14 +50,16 @@ trait KafkaMetricsReporter {
 object KafkaMetricsReporter {
   val ReporterStarted: AtomicBoolean = new AtomicBoolean(false)
 
-  def startReporters (verifiableProps: VerifiableProperties) {
+  def startReporters (verifiableProps: VerifiableProperties):List[KafkaMetricsReporter] = {
+    var reporters = List[KafkaMetricsReporter]()
```

Comment:
This should probably be inside the `synchronized` block. I think we should probably return `Option[Seq[KafkaMetricsReporter]]`. `None` would be returned if the start had already been done. An alternative is to store the reporters in a field and return the stored ones in that case.